src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,363 @@
+                attemptUpdate(existingEntry, draft);
+            } else {
+                attemptCreate(draft);
+            }
+            statistics.incrementProcessed();
+        });
+    }
+
+    /**
+     * Returns mapping of {@link SkuKeyTuple} to {@link InventoryEntry} of instances existing in a database.
+     * Instances are fetched from API by skus, that corresponds to skus present in {@code drafts}.
+     *
+     * @param drafts {@link List} of drafts
+     * @return {@link SkuKeyTuple} to {@link InventoryEntry} of existing entries that correspond to passed
+     * {@code drafts} by sku comparision.
+     */
+    private Map<SkuKeyTuple, InventoryEntry> fetchExistingInventories(final List<InventoryEntryDraft> drafts) {
+        final Set<String> skus = extractSkus(drafts);
+        final List<InventoryEntry> existingEntries = inventoryService.fetchInventoryEntriesBySkus(skus);
+        return existingEntries.stream()
+                .collect(toMap(SkuKeyTuple::of, entry -> entry));
+    }
+
+    /**
+     *
+     * @param inventories {@link List} of {@link InventoryEntryDraft} where each draft contains its sku
+     * @return {@link Set} of distinct skus found in {@code inventories}
+     */
+    private Set<String> extractSkus(final List<InventoryEntryDraft> inventories) {
+        return inventories.stream()
+                .map(InventoryEntryDraft::getSku)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Tries to update {@code entry} in CT platform with data from {@code draft}.
+     * It calculates list of {@link UpdateAction} and calls API only when there is a need.
+     * Before calculate differences, the channel reference from {@code draft} is replaced, so it points to
+     * proper channel ID in target system.
+     * It either updates inventory entry and increments "updated" counter in statistics, or increments "failed" counter
+     * and log error in case of any exception.
+     *
+     * @param entry entry from existing system that could be updated.
+     * @param draft draft containing data that could differ from data in {@code entry}.
+     *              <strong>Sku isn't compared</strong>
+     * @return {@link CompletionStage} that contains either updated {@link InventoryEntry} when succeeded or
+     * {@code null} otherwise
+     */
+    private CompletionStage<InventoryEntry> attemptUpdate(final InventoryEntry entry, final InventoryEntryDraft draft) {
+        final Optional<InventoryEntryDraft> fixedDraft = replaceChannelReference(draft);
+        if (fixedDraft.isPresent()) {
+            final List<UpdateAction<InventoryEntry>> updateActions =
+                    InventorySyncUtils.buildActions(entry, fixedDraft.get(), syncOptions, typeService);
+            if (!updateActions.isEmpty()) {
+                return inventoryService.updateInventoryEntry(entry, updateActions)
+                        .handle((updatedEntry, exception) -> {
```

Comment:
I'll adjust this when https://github.com/commercetools/commercetools-sync-java/pull/37 will be merged. Then CompletionStages would be collected and returned as one CompletionStage of entire process. This should fix the issue, right?